pkg/asset/manifests/cloudproviderconfig.go

```diff
@@ -152,8 +152,7 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 	case vspheretypes.Name:
 		folderPath := installConfig.Config.Platform.VSphere.Folder
 		if len(folderPath) == 0 {
-			dataCenter := installConfig.Config.Platform.VSphere.Datacenter
-			folderPath = fmt.Sprintf("/%s/vm/%s", dataCenter, clusterID.InfraID)
+			folderPath = fmt.Sprintf("%s/vm/%s", vspheretypes.FullyQualifiedPath(installConfig.Config.Platform.VSphere.Datacenter), clusterID.InfraID)
```

Comment:
nit: I think this line is too long. For readability I would avoid calling a function as an arg

```suggestion
                        dataCenter := vspheretypes.FullyQualifiedPath(installConfig.Config.Platform.VSphere.Datacenter)
			folderPath = fmt.Sprintf("%s/vm/%s", dataCenter, clusterID.InfraID)
```